src/Adapter/Product/Stock/QueryHandler/GetStockMovementHistoriesHandler.php

```diff
@@ -0,0 +1,138 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Stock\QueryHandler;
+
+use DateTime;
+use DateTimeImmutable;
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Repository\StockAvailableRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Repository\StockMovementFilter;
+use PrestaShop\PrestaShop\Adapter\Product\Stock\Repository\StockMovementRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Query\GetStockMovementHistories;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\QueryHandler\GetStockMovementHistoriesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\QueryResult\RangeStockMovementHistory;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\QueryResult\SingleStockMovementHistory;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\QueryResult\StockMovement;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\QueryResult\StockMovementHistory;
+
+class GetStockMovementHistoriesHandler implements GetStockMovementHistoriesHandlerInterface
+{
+    /**
+     * @var StockAvailableRepository
+     */
+    private $stockAvailableRepository;
+
+    /**
+     * @var StockMovementRepository
+     */
+    private $stockMovementRepository;
+
+    public function __construct(
+        StockAvailableRepository $stockAvailableRepository,
+        StockMovementRepository $stockMovementRepository
+    ) {
+        $this->stockAvailableRepository = $stockAvailableRepository;
+        $this->stockMovementRepository = $stockMovementRepository;
+    }
+
+    public function handle(GetStockMovementHistories $query): array
+    {
+        $stockId = $this->stockAvailableRepository->getStockIdByProduct($query->getProductId());
+        $mainFilter = (new StockMovementFilter())->setStockIds($stockId);
+        $rangeFilter = (new StockMovementFilter())->setIsOrder(false);
+
+        return array_map(
+            function (array $historyRow): StockMovementHistory {
+                return $this->createStockMovementHistory($historyRow);
+            },
+            $this->stockMovementRepository->getLastStockMovementHistories(
+                $mainFilter,
+                $rangeFilter,
+                $query->getOffset(),
+                $query->getLimit()
+            )
+        );
+    }
+
+    protected function createStockMovementHistory(array $historyRow): StockMovementHistory
+    {
+        if (1 === (int) $historyRow['id_stock_mvt_count']) {
```

Comment:
We have a debate about yoda conditions https://github.com/PrestaShop/PrestaShop/issues/26405

Reply:
I got use to yoda style only because it recently became a common code style constraint. But I totally dislike this practice, totally useless in my humble opinion.
I commented the debate: https://github.com/PrestaShop/PrestaShop/issues/26405#issuecomment-1018309602

Reply:
I agree with your opinion about yoda style. So do we revert this line?

```
        if ((int) $historyRow['id_stock_mvt_count'] === 1) {

```

Reply:
Already reverted, with relief :))